src/shell/store/models.js

```diff
@@ -99,15 +99,21 @@ export function fetchModel(modelZUID) {
             type: "FETCH_MODEL_SUCCESS",
             payload: res.data,
           });
+        } else if (res.status === 404) {
```

Comment:
See https://github.com/zesty-io/manager-ui/issues/1268#issuecomment-1257320449 for a description of the problem.

> Not sure why we need this specific 404 case to do nothing. This block is not even throwing an error

I do not believe that we want to throw an error. It is totally valid for a model fetch to 404 in the case where a model has been deleted. The Activity Log works correctly in this case, except a warning notification is given to the user and an error is sent to Sentry. I think it will confuse the user to give them an error notification when the page is working as intended.

> this could potentially be a breaking change if a consumer is relying on the promise rejection

The only consumers of this function are the Activity Log and the Dashboard, neither of which rely on the promise rejection.

> The else already catches this and properly throws an error

I do not believe that we want to throw an error on a 404. If you look at `fetchModels`, `fetchField` and `fetchFields`, none of them throw on a 404. I think that is the pattern that should be followed.